app/controllers/application_controller.rb

```diff
@@ -171,6 +171,7 @@ def sign_in(user, remember_me: nil)
     user.sign_ins.create(
       ip: user_ip, country: country_from_ip, user_agent: request.user_agent
     )
+    user.touch(:last_sign_in_at)
```

Comment:
think its worth refactoring the sign_in creation and touch to `User#record_sign_in`

Reply:
I don't understand. Are you talking about `User::SignIn` creation? Also `User#record_sign_in` doesn't exist?

Reply:
Yeah, so basically add:

```rb
class User
  def record_sign_in(args)
    sign_ins.create(args)
    user.touch(:last_sign_in_at)
  end
end
```

Then in the controller we can just call:

```diff
- user.sign_ins.create(
-   ip: user_ip, country: country_from_ip, user_agent: request.user_agent
- )
+ user.record_sign_in(ip: user_ip, country: country_from_ip, user_agent: request.user_agent)
```

Was just thinking it would keep the controller a bit cleaner, and we could add a test for the model level method to assert that it touches the record and creates/doesn't create the sign in depending on whether retention is enabled.